src/Workload/Microsoft.Maui.Sdk/Sdk/BundledVersions.in.targets

```diff
@@ -14,49 +19,57 @@
-        DefaultRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
-        LatestRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
         TargetingPackName="Microsoft.Maui.Core.Ref.$(_MauiPlatformName)"
-        TargetingPackVersion="$(_MauiRuntimePackVersion)"
         RuntimePackNamePatterns="Microsoft.Maui.Core.Runtime.**RID**"
         RuntimePackRuntimeIdentifiers="$(_MauiPlatformName)"
         Profile="$(TargetPlatformIdentifier)"
         RuntimePackAlwaysCopyLocal="$(_MauiRuntimePackAlwaysCopyLocal)"
     />
-    <KnownFrameworkReference
+    <_MauiKnownFrameworkReference
         Condition=" '$(UseMaui)' == 'true' "
         Include="Microsoft.Maui.Controls"
-        TargetFramework="$(MauiWorkloadTargetFramework)"
         RuntimeFrameworkName="Microsoft.Maui.Controls"
-        DefaultRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
-        LatestRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
         TargetingPackName="Microsoft.Maui.Controls.Ref.$(_MauiPlatformName)"
-        TargetingPackVersion="$(_MauiRuntimePackVersion)"
         RuntimePackNamePatterns="Microsoft.Maui.Controls.Runtime.**RID**"
         RuntimePackRuntimeIdentifiers="$(_MauiPlatformName)"
         Profile="$(TargetPlatformIdentifier)"
         RuntimePackAlwaysCopyLocal="$(_MauiRuntimePackAlwaysCopyLocal)"
     />
-    <KnownFrameworkReference
+    <_MauiKnownFrameworkReference
         Condition=" '$(UseMaui)' == 'true' or '$(UseMauiEssentials)' == 'true' "
         Include="Microsoft.Maui.Essentials"
-        TargetFramework="$(MauiWorkloadTargetFramework)"
         RuntimeFrameworkName="Microsoft.Maui.Essentials"
-        DefaultRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
-        LatestRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
         TargetingPackName="Microsoft.Maui.Essentials.Ref.$(_MauiPlatformName)"
-        TargetingPackVersion="$(_MauiRuntimePackVersion)"
         RuntimePackNamePatterns="Microsoft.Maui.Essentials.Runtime.**RID**"
         RuntimePackRuntimeIdentifiers="$(_MauiPlatformName)"
         Profile="$(TargetPlatformIdentifier)"
         RuntimePackAlwaysCopyLocal="$(_MauiRuntimePackAlwaysCopyLocal)"
     />
+
+    <!-- Current SDK version -->
+    <KnownFrameworkReference
+        Include="@(_MauiKnownFrameworkReference)"
+        TargetFramework="$(MauiWorkloadTargetFramework)"
+        DefaultRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
+        LatestRuntimeFrameworkVersion="$(_MauiRuntimePackVersion)"
+        TargetingPackVersion="$(_MauiRuntimePackVersion)"
+    />
+
+    <!-- Previous SDK version -->
+    <KnownFrameworkReference
+        Include="@(_MauiKnownFrameworkReference)"
+        TargetFramework="$(_MauiPreviousWorkloadTargetFramework)"
+        DefaultRuntimeFrameworkVersion="$(_MauiPreviousRuntimePackVersion)"
+        LatestRuntimeFrameworkVersion="$(_MauiPreviousRuntimePackVersion)"
+        TargetingPackVersion="$(_MauiPreviousRuntimePackVersion)"
+    />
```

Comment:
@jonathanpeppers  is this bad or good? I see the core SDK lists all versions here: https://github.com/dotnet/installer/blob/34ba3a68bb4cee2bb0a71e1b2ee6038ba44a5cd1/src/redist/targets/GenerateBundledVersions.targets#L489

So I copied them. When building with net6 TFMs it does not use this file so it does not matter, but when we build net7 TFMs we get both... Then the build selects the correct one. Is this good or bad? Why does .NET use all the versions? 

@dsplaisted do you have any words of wisdom?

Reply:
The .NET SDK supports targeting all previous versions of .NET, so it has the known framework references for .NET 7, 6, 5, and .NET Core 3.1 and 3.0 (prior versions didn't use framework references).

If you have a separate workload for Maui support for .NET 6 and .NET 7, probably the .NET 7 workload should only have the known framework references for .NET 7, and rely on the .NET 6 workload to provide the known framework references for .NET 6.

Reply:
That is good to know, thanks!